src/services/services.ts

```diff
@@ -6068,18 +6102,36 @@ namespace ts {
                 // The search set contains at least the current symbol
                 let result = [symbol];
 
-                // If the symbol is an alias, add what it aliases to the list
-                if (isImportSpecifierSymbol(symbol)) {
-                     result.push(typeChecker.getAliasedSymbol(symbol));
+                // If the location is name of property symbol from object literal destructuring pattern
+                // Search the property symbol
+                //      for ( { property: p2 } of elems) { }
+                if (isNameOfPropertyAssignment(location) &&
+                    location.parent.kind !== SyntaxKind.ShorthandPropertyAssignment &&
+                    isArrayLiteralOrObjectLiteralDestructuringPattern(location.parent.parent)) {
+                    const typeOfObjectLiteral = typeChecker.getTypeOfArrayLiteralOrObjectLiteralDestructuringAssignment(<Expression>location.parent.parent);

```

Comment:
You jsut need the symbol here, so i would say this should return a symbol and possibly be named: `getSoruceSymbolOfBindingPatternAssignment`